src/host.rs

```diff
@@ -192,6 +192,15 @@ impl<'a> HostAndPort<&'a str> {
     }
 }
 
+impl<S: AsRef<str>> fmt::Display for HostAndPort<S> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        try!(self.host.fmt(f));
```

Comment:
what about using the "new" ? operator?

~~~
self.host.fmt(f)?;
self.write_str(":")?;
self.port.fmt(f)
~~~